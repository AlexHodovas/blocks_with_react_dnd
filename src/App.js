import React from 'react';
import PropTypes from 'prop-types';
import Table from './components/Table';
import Wrapper from './components/Wrapper';

const Wrapper2 = ({ initialWidth, initialHeight, cellSize }) => (
  <Wrapper>
    <Table
      initialWidth={initialWidth}
      initialHeight={initialHeight}
      cellSize={cellSize}
    />
  </Wrapper>
)

const App = () => <>
  <Wrapper2
    initialWidth={4}
    initialHeight={4}
    cellSize={50}
  />
</>

Wrapper2.propTypes = {
  initialWidth: PropTypes.number.isRequired, 
  initialHeight: PropTypes.number.isRequired, 
  cellSize: PropTypes.number.isRequired,
};

export default App;