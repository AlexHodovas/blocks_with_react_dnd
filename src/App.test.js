import React from 'react';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import App from './App';

configure({ adapter: new Adapter() });

it('renders correctly', () => {
  const tree = renderer
    .create(
      <App 
        initialWidth={4}
        initialHeight={4}
        cellSize={50}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

describe('buttons', () => {
  const onClick = jest.fn();
  let wrapper;

  beforeEach(() => {
    wrapper = mount(<App onClick={onClick} /> );
  })

  it('addRowButton works', () => {
    const button = wrapper.find('#addRowButton');    
    button.simulate('click')
    expect(wrapper.find('#addRowButton').props().tableHeight).toEqual(5);
  })

  it('addColumnButton works', () => {
    const button = wrapper.find('#addColumnButton');
    button.simulate('click')
    expect(wrapper.find('#addColumnButton').props().tableWidth).toEqual(5);
  })

  it('deleteColumnButton works', () => {
    const button = wrapper.find('#deleteColumnButton');    
    button.simulate('click')
    expect(wrapper.find('#deleteColumnButton').props().tableWidth).toEqual(3);
  })

  it('deleteColumnButton works', () => {
    const button = wrapper.find('#deleteColumnButton');    
    button.simulate('click')
    expect(wrapper.find('#deleteColumnButton').props().tableWidth).toEqual(3);
  })

  it('last row not removed', () => {
    const button = wrapper.find('#deleteRowButton');    
    button.simulate('click')
    button.simulate('click')
    button.simulate('click')
    expect(wrapper.find('#deleteRowButton').props().isOnlyOneRow()).toEqual(true);
    expect(wrapper.find('#deleteRowButton').props().isDeleteRowButtonHidden).toEqual(true);
  })

  it('last column not removed', () => {
    const button = wrapper.find('#deleteColumnButton');    
    button.simulate('click')
    button.simulate('click')
    button.simulate('click')
    expect(wrapper.find('#deleteColumnButton').props().isOnlyOneColumn()).toEqual(true);
    expect(wrapper.find('#deleteColumnButton').props().isDeleteColumnButtonHidden).toEqual(true);
  })

});
