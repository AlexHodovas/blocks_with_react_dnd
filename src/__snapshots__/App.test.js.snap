// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-AxiKw gMkJnZ"
>
  <table
    className="sc-AxheI dTrmKq"
    onMouseEnter={[Function]}
  >
    <thead />
    <tbody>
      <tr>
        <td
          className="sc-AxgMl esozLi"
          onMouseOver={[Function]}
        />
        <td
          className="sc-AxgMl esozLi"
          onMouseOver={[Function]}
        />
        <td
          className="sc-AxgMl esozLi"
          onMouseOver={[Function]}
        />
        <td
          className="sc-AxgMl esozLi"
          onMouseOver={[Function]}
        />
      </tr>
      <tr>
        <td
          className="sc-AxgMl esozLi"
          onMouseOver={[Function]}
        />
        <td
          className="sc-AxgMl esozLi"
          onMouseOver={[Function]}
        />
        <td
          className="sc-AxgMl esozLi"
          onMouseOver={[Function]}
        />
        <td
          className="sc-AxgMl esozLi"
          onMouseOver={[Function]}
        />
      </tr>
      <tr>
        <td
          className="sc-AxgMl esozLi"
          onMouseOver={[Function]}
        />
        <td
          className="sc-AxgMl esozLi"
          onMouseOver={[Function]}
        />
        <td
          className="sc-AxgMl esozLi"
          onMouseOver={[Function]}
        />
        <td
          className="sc-AxgMl esozLi"
          onMouseOver={[Function]}
        />
      </tr>
      <tr>
        <td
          className="sc-AxgMl esozLi"
          onMouseOver={[Function]}
        />
        <td
          className="sc-AxgMl esozLi"
          onMouseOver={[Function]}
        />
        <td
          className="sc-AxgMl esozLi"
          onMouseOver={[Function]}
        />
        <td
          className="sc-AxgMl esozLi"
          onMouseOver={[Function]}
        />
      </tr>
    </tbody>
  </table>
  <div
    className="sc-AxjAm jGrzUd"
  >
    <div
      onClick={[Function]}
      style={
        Object {
          "height": "48px",
          "lineHeight": "48px",
          "textAlign": "center",
        }
      }
    >
      +
    </div>
  </div>
  <div
    className="sc-AxirZ QCVaJ"
  >
    <div
      onClick={[Function]}
      style={
        Object {
          "height": "48px",
          "lineHeight": "48px",
          "textAlign": "center",
        }
      }
    >
      +
    </div>
  </div>
  <div
    className="sc-AxhUy bGMevz"
    onClick={[Function]}
  >
    <div
      style={
        Object {
          "height": "48px",
          "lineHeight": "48px",
          "textAlign": "center",
        }
      }
    >
      -
    </div>
  </div>
  <div
    className="sc-AxhCb eCtuRO"
    onClick={[Function]}
  >
    <div
      style={
        Object {
          "height": "48px",
          "lineHeight": "48px",
          "textAlign": "center",
        }
      }
    >
      -
    </div>
  </div>
</div>
`;
